extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -573,16 +576,8 @@ function processTextField(context: FieldContext): void {
 }
 
 function processPasswordField(context: FieldContext): void {
-	const passwordLabel = createLabel(context.view, { text: context.fieldInfo.label, description: context.fieldInfo.description, required: context.fieldInfo.required, width: context.fieldInfo.labelWidth, cssStyles: context.fieldInfo.labelCSSStyles });
-	const passwordInput = context.view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
-		ariaLabel: context.fieldInfo.label,
-		inputType: 'password',
-		required: context.fieldInfo.required,
-		placeHolder: context.fieldInfo.placeHolder,
-		width: context.fieldInfo.inputWidth
-	}).component();
-	context.onNewInputComponentCreated(context.fieldInfo.variableName!, { component: passwordInput, isPassword: true });
-	addLabelInputPairToContainer(context.view, context.components, passwordLabel, passwordInput, context.fieldInfo);
+	processTextField(context);
+	const passwordInput = context.inputComponents[context.fieldInfo.variableName!].component as azdata.InputBoxComponent;
 
 	if (context.fieldInfo.type === FieldType.SQLPassword) {
 		const invalidPasswordMessage = getInvalidSQLPasswordMessage(context.fieldInfo.label);
```

Comment:
If someone gets some free time it'd be nice to switch this over to using the text validation as well

Reply:
do you mean get rid of sql_password field type?

Reply:
I think Alan just did and my other 3 part PR for new validations also does the same.

Reply:
I will do it after we switch to the new validation work that I am putting in place because syntax of specifying the regex validation changes with that PR.

Reply:
But let us continue the discussion here on what we would want that experience to be.